app/serializers/event_serializer.rb

```diff
@@ -8,6 +8,7 @@ class EventSerializer < ActiveModel::Serializer
   attribute :pt_longitude, key: :longitude
   attribute :event_name, key: :name
   attribute :service_description, key: :service
+  attribute :distance
```

Comment:
I don't really like the idea of adding the `distance` field into the json response. As it stands, measuring the distance from centroid of the zip, at best, a bad approximation of the actual distance from the user. At worst it's wildly inaccurate. Surfacing this data to the user will just reduce confidence in the api.

I agree that distance is an important metric, but with our current ability to calculate it I would prefer to do the estimated sort by distance in the backend and return the sorted list to the ui rather than rely on the ui to sort by our distance values. Keep in mind that right now the only consumer of this api is our team's frontend, but that may not always be the case. It would not be unreasonable to open up this api to other agencies/applications so we should be conscientious about what data we add to the response.

Thoughts?

Reply:
David and I discussed the zip centroid issue earlier this week.  We do not
plan on displaying the distance based on zip code centroid in the UI due to
the accuracy viz a viz the user’s location you mentioned, but would sort
events for a particular event_date by distance.  The usability issue this
addresses applies to the case where searching a licking county zip like
40446 orders the results in almost the reverse order of distance:

http://localhost:8888/api/agencies?zip_code=43046

1. Grove City (4323) - 27.71 miles
2. Buckeye Lake (43008)  - 3.44 miles
3. Jacksontown (43030) - 7.37 miles
4. Millersport (43046) - .3 miles

Sorting events by event_dates by distance would work out well for this
case.  It would make little difference which event displays first in the
case where the events are in the same zip_code.  The boundary cases might
pose a problem but I am not sure how accurate we can be in those cases
based on zip_code centroid or some other form of estimation.

The structure of the api results is agency->event->event_date.  The UI
structure is event_date->(events+agencies).  I don’t know if we could sort
on the back end that matches what the UI needs in the way they currently
process the results but we could discuss with the UI team.

Regardless of who sorts when we get to the point where the UI sends
lat/long for the user’s location I think we will want to display distance
in the results.

I think proper API documentation could differentiate between zip code
centroid distance (or some other estimation) and precise geo coordinate
distance.

Reply:
Let me add a few general comments, then I'll try to address some specifics:

#1 - We want to sort the information within a date on distance, closest to farthest. 
Customer story: I am a customer, I input my zip code and am looking for a food resource that is open and close to me. 

The specific method we use to achieve this result is up for debate, and I'll try and comment on approach:

> @nvengal measuring the distance from centroid of the zip, at best, a bad approximation of the actual distance from the user. At worst it's wildly inaccurate. Surfacing this data to the user will just reduce confidence in the api.

Excellent comment. As much as we want to quickly improve the Agency Finder tool, we want it to serve up accurate information more. I think we are all in agreement, **if we use the zipcode centroid lat/long - agency lat/long to calculate distance we do not want to display that information to the user**.

>  I would prefer to do the estimated sort by distance in the backend and return the sorted list to the ui rather than rely on the ui to sort by our distance values.

This seems like a logical response, and from what I understand of the Front end, this should work. I also like the idea that we aren't putting information out that could potentially confuse a user of the API. @seanlawrenz could you comment on the Front End's use of the JSON? Does the order the agencies occur in the JSON result in how they are painted into the Page? If not, what information would you like the in the JSON to achieve the desired result?

> @PhilNorman2 Regardless of who sorts when we get to the point where the UI sends
lat/long for the user’s location I think we will want to display distance
in the results.

Yep! 100%. I think we are all in agreement. Thanks for sharing some thoughts from our other conversation.